cmd/odf/set/osd_settings.go

```diff
@@ -0,0 +1,52 @@
+package set
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/red-hat-storage/odf-cli/cmd/odf/root"
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	"github.com/spf13/cobra"
+)
+
+var osdFullRatioCmd = &cobra.Command{
+	Use:                "set-full-ratio",
+	Short:              "Configure ceph osd 'set-full-ratio' setting",
+	DisableFlagParsing: true,
+	Args:               cobra.ExactArgs(1),
+	Example:            "odf set ceph osd set-full-ratio 0.9",
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := cmd.Context()
+		err := validateArgs(args[0])
+		if err != nil {
+			logging.Fatal(err)
+		}
+
+		cephArgs := []string{"osd", "set-full-ratio", args[0]}
+		out, err := exec.RunCommandInOperatorPod(ctx, root.ClientSets, "ceph", cephArgs, root.OperatorNamespace, root.StorageClusterNamespace, true)
+		if err != nil {
+			logging.Fatal(fmt.Errorf("failed to run command ceph with args %v: %v", cephArgs, err))
+		}
+		logging.Info(out)
+	},
+}
+
+func validateArgs(arg string) error {
```

Comment:
What error does ceph give if an invalid is passed and we don't validate it? Perhaps ceph's checks are sufficient and we don't need to validate?

Reply:
right we don't need to validate
```bash
./bin/odf -n rook-ceph set ceph osd set-full-ratio 2.0
Error: failed to run command ceph with args [osd set-full-ratio 2.0]: Invalid command: 2.0 not in range [0.0, 1.0]
osd set-full-ratio <ratio:float> :  set usage ratio at which OSDs are marked full
Error EINVAL: invalid command
. failed to run command. command terminated with exit code 22
```